File: api/Negotiate.fs
namespace Api

open Microsoft.Azure.Functions.Worker
open Microsoft.Extensions.Logging

//module Negotiate =
    
//    [<Function("Negotiate")>]
//    let run 
//        ([<HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)>] req: HttpRequest) 
//        ([<SignalRConnectionInfo(HubName = "serverless")>] connectionInfo: SignalRConnectionInfo)
//        (log: ILogger) =
//            connectionInfo

File: api/Negotiate.cs
using System;
using System.IO;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Http;
using Microsoft.AspNetCore.Http;
using Microsoft.Azure.WebJobs.Extensions.SignalRService;
using Microsoft.Azure.WebJobs.Host;
using Microsoft.Extensions.Logging;
using Newtonsoft.Json;

namespace Api;

public static class Negotiate
{
    [FunctionName("negotiate")]
    public static SignalRConnectionInfo Run(
        [HttpTrigger(AuthorizationLevel.Anonymous)]HttpRequest req,
        [SignalRConnectionInfo(HubName = "chat")]SignalRConnectionInfo connectionInfo)
    {
        return connectionInfo;
    }
}